### PR TITLE
lib: Update FatFs to R0.11.

### PR DESCRIPTION
**Pull request by danicampora**
There are lots of cosmetic changes, but this release brings a  very
important bug fix:
- Fixed f_unlink() does not remove cluster chain of the file.

With R0.10c if you try to write a file that is too large to fit in the
free space of the drive, the operation fails, you delete the incomplete
file, and it seems to be erased, but the space is not really freed,
because any subsequent write operations fail because the drive is
"still" full. The only way to recover from this is by formatting the
drive. 
I can confirm that R0.11 fixes the problem.

### CODE REVIEW

File: lib/fatfs/ff.c [hunk 1]
```diff
@@ -562,12 +458,6 @@ static const BYTE ExCvt[] = _EXCVT;	/* Upper conversion table for extended chara
 /* String functions                                                      */
 /*-----------------------------------------------------------------------*/
 
-// dpgeorge:
-// We use stdlib version of memset and memcmp to reduce code size.
-// We use the original custom version of mem_cpy so that gcc doesn't
-// recognise the builtin function and then inline it.  Allowing gcc
-// to use the builtin memcpy increases code size by 72 bytes.
```
Comment (dpgeorge) [inline lib/fatfs/ff.c]: Why did you remove this comment, and the #if's below?

**Comment by danicampora** [comment]
Sorry, I completely overlooked it :-(, let me fix it...

**Comment by danicampora** [comment]
@dpgeorge fixed, should be fine now. I added a few more comments with your name to make the custom modifications more visible.

**Comment by dpgeorge** [comment]
Please add history.txt.

**Comment by danicampora** [comment]
OK, just did it ;-)

**Comment by dpgeorge** [comment]
Great, thanks!

Will merge after dealing with #1170.

**Comment by dpgeorge** [comment]
Rebased and merged in 7b19e99edd2ca5d60154ffaa6007c936f274a15a.
